### PR TITLE
PIPE2D-1427: fitFluxCal.py: Address problems arising from absent arms.

### PR DESCRIPTION
**Pull request by SogoMineo**
If an arm is absent, spectrum in its wavelength range is not available. Because we compare broadband fluxes with integrations of spectra, sampling points absent from the integration ranges lead to erroneous chi^2.

To address this problem, we throw away broadband fluxes if the bands are hardly covered by the observed spectra. In addition, we interpolate or extrapolate the observed spectra so that they will cover entire bands of broadband filters.

### CODE REVIEW

**Review by PaulPrice — APPROVED** [review]
I think the integration test failed.

File: python/pfs/drp/stella/fitFluxCal.py [hunk 1]
```diff
@@ -378,6 +389,85 @@ def _getFilterCurve(self, filterName) -> FilterCurve:
...
+            Spectrum.
+        bbFlux : `List[Tuple[float, float, str]]`
+            Broadband fluxes. Each element of the list is a tuple of
+            ``(flux, fluxErr, filterName)``
+        arms : `str`
+            Existent arms. "brn" and "bmn" for example.
+        """
+        if not all(filterName.endswith("_ps1") for flux, fluxErr, filterName in bbFlux):
+            if self.log:
+                self.log.warning("Flux calibration cannot be done with non-PS1 broadband fluxes.")
```
[the remark this answers is not in the document]
Comment (SogoMineo) [reply python/pfs/drp/stella/fitFluxCal.py:419]: Yes. I also asked the question to Tanaka-san, and he thinks that we can postpone the issue until such fluxes actually come.

File: python/pfs/drp/stella/fitFluxCal.py [hunk 2]
```diff
@@ -587,6 +677,117 @@ def objective3(params: np.ndarray) -> float:
...
+        Wavelength range (nm) in which to compute ``flux1``.
+    wlRange2 : `Tuple[float, float]`
+        Wavelength range (nm) in which to compute ``flux2``.
+    """
+    flux_isfinite = np.isfinite(spectrum.flux)
+    index1 = flux_isfinite & (wlRange1[0] < spectrum.wavelength) & (spectrum.wavelength < wlRange1[1])
+    index2 = flux_isfinite & (wlRange2[0] < spectrum.wavelength) & (spectrum.wavelength < wlRange2[1])
+
+    wl1 = np.median(spectrum.wavelength[index1])
+    wl2 = np.median(spectrum.wavelength[index2])
```
[the remark this answers is not in the document]
Comment (SogoMineo) [reply python/pfs/drp/stella/fitFluxCal.py:703]: These must not be `mean`. There may be absorption lines. I want the continuum only.

File: python/pfs/drp/stella/fitFluxCal.py [hunk 3]
```diff
@@ -587,6 +677,117 @@ def objective3(params: np.ndarray) -> float:
...
+    spectrum : `PfsSingle`
+        Spectrum.
+    wlRange1 : `Tuple[float, float]`
+        Wavelength range (nm) in which to compute ``flux1``.
+    wlRange2 : `Tuple[float, float]`
+        Wavelength range (nm) in which to compute ``flux2``.
+    """
+    flux_isfinite = np.isfinite(spectrum.flux)
+    index1 = flux_isfinite & (wlRange1[0] < spectrum.wavelength) & (spectrum.wavelength < wlRange1[1])
+    index2 = flux_isfinite & (wlRange2[0] < spectrum.wavelength) & (spectrum.wavelength < wlRange2[1])
```
[the remark this answers is not in the document]
Comment (SogoMineo) [reply python/pfs/drp/stella/fitFluxCal.py:700]: Do masks count when I have to use median (not mean) below?

**Comment by SogoMineo** [comment]
I pushed changes that I made according to the comments. I have not added a mask check yet. I am not sure I have to check it when I want just median.

File: python/pfs/drp/stella/fitFluxCal.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (PaulPrice) [reply python/pfs/drp/stella/fitFluxCal.py]: The wavelength array is sampled regularly and has no outliers.

File: python/pfs/drp/stella/fitFluxCal.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (PaulPrice) [reply python/pfs/drp/stella/fitFluxCal.py]: The median gives you some protection against outliers, but why not use the information we already have available to make it more robust?

[the remark this answers is not in the document]
Comment (SogoMineo) [reply python/pfs/drp/stella/fitFluxCal.py]: I added a mask check here. (I had not wanted to pull `config.badMask` all the way from a far.)

**Pull request merged**